### ISSUE TITLE
Assets file should use relative paths to enable moving projects more easily

### ISSUE THREAD

**Issue by davkean** [NuGet/Home#4582]
Due to https://github.com/dotnet/roslyn-project-system/issues/1538, moving a project on disk results in a project with broken packages/IntelliSense when you next open it.

To avoid this situation until we reload targets automatically in the future, NuGet should burn in a value in ProjectAssetsFile that is relative to the intermediate directory. This will make it less likely developers will run into this.

**Comment by emgarten** [comment]
For ProjectAssetsFile I agree this path should be relative.

For the rest of the paths, the issue is that the global user packages folder could be anywhere relative to where the project lives, and there could also be fallback folders that are needed.

Changing these files to use relative paths would only make it more confusing because it would seem like these are suppose to be portable. Users need to restore after moving projects and should avoid checking these into source control.

**Comment by davkean** [comment]
I'm not sure how it would result in more confusion - right now you need to _restart_ Visual Studio for your references to start working. This has nothing to do with source control, this is literally copy and pasting your project to a new directory.

**Comment by nkolev92** [comment]
Reopening as the last fix has been reverted because it broke .NET Core builds.

**Comment by emgarten** [comment]
@nkolev92 @rohit21agrawal would you provide some details on what actually broke this? There issue blocking this relative path change from going in should be linked here.

**Comment by rohit21agrawal** [comment]
@emgarten here are the repro steps:

1) Use a VSIX that has this change and install it to VS2017
2) Create new .NET Core project
3) Restore in VS
4) Build in VS

Build will fail.

**Comment by emgarten** [comment]
The only change here is to make a property value a relative path instead of an absolute path. NuGet does not read this property, so if this doesn't work there should be an issue filed on the consumer of this property.

**Comment by nkolev92** [comment]
Self-assigning so I can do some investigation on what changes would be required on SDK side.

**Comment by nkolev92** [comment]
Look at me go, I finally caught up to this :D

I borrowed the changes that Justin made last time he attempted to fix this and I investigated "why" this is happening. 

The root cause of all these problems is the fact that the related targets that depend on the `$(ProjectAssetsFile)` completely fail. 

There are a lot of usages of this throughout the SDK targets. A lot of which assume a full path. 

I have decided to proceed with https://github.com/dotnet/project-system/issues/1538#issuecomment-483465834

```xml
    <ProjectAssetsFile Condition=" '$(ProjectAssetsFile)' == '' ">$(MSBuildThisFileDirectory)project.assets.json</ProjectAssetsFile>
```

PR is out at https://github.com/NuGet/NuGet.Client/pull/2805.
